### PR TITLE
Automate the port forwarding for the Communicator

### PR DESCRIPTION
**Pull request by bheuvel**
Automate the creation of the public port (randomized)
Automate detection of the guest port

Based on the issue #85

### CODE REVIEW

File: lib/vagrant-cloudstack/action/run_instance.rb [hunk 1]
```diff
@@ -200,7 +210,28 @@ def call(env)
               :privateport  => pf_private_port,
               :openfirewall => pf_open_firewall
             }
-            create_port_forwarding_rule(env, port_forwarding_rule)
+            retryable(:on => DuplicatePFRule, :tries => 10) do
```
Comment (miguelaferreira) [inline lib/vagrant-cloudstack/action/run_instance.rb:215]: Awesome!

File: lib/vagrant-cloudstack/exceptions/exceptions.rb [hunk 2]
```diff
@@ -3,6 +3,8 @@ module Cloudstack
     module Exceptions
       class IpNotFoundException < Exception
       end
+      class DuplicatePFRule < Exception
+  	  end
```
Comment (miguelaferreira) [inline lib/vagrant-cloudstack/exceptions/exceptions.rb]: Indentation...

**Comment by j00p34** [comment]
Tested:
multiple linux servers running. random ports are working like a charm.

**Comment by bheuvel** [comment]
@j00p34 thanks!. Just (now) rebased this PR to resolve the mergeconflicts.

**Comment by j00p34** [comment]
re-tested rebased code. Now with Linux (2x) and windows ;-) (1x) :+1:

**Pull request merged**